### PR TITLE
feat(semantic-release): add semantic-release

### LINKED ISSUE #16: Adding Semantic Release
### Description:
1. Adding Semantic release to the project

### Outcome:
1. Manual release of the package

### PR DESCRIPTION
**Pull request by niloysikdar**
### Outcome:
1. add semantic-release
2. add `yarn release` script
3. add @semantic-release/commit-analyzer
4. add @semantic-release/release-notes-generator
5. add @semantic-release/changelog config
6. add @semantic-release/github
7. add @semantic-release/npm
8. add @semantic-release/git config

fix #16

Signed-off-by: Niloy Sikdar <niloysikdar30@gmail.com>

### CODE REVIEW

File: .releaserc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "@semantic-release/release-notes-generator",
+    [
+      "@semantic-release/changelog",
+      {
+        "changelogFile": "CHANGELOG.md"
+      }
+    ],
+    "@semantic-release/npm",
+    [
+      "@semantic-release/git",
```
Comment (agentmilindu) [inline .releaserc:14]: Don't we need the GitHub plugin to create a GitHub release?

**Review by agentmilindu — CHANGES_REQUESTED** [review]

File: .releaserc (hunk 1 above)
Comment (niloysikdar) [reply .releaserc:14]: If we also need the GitHub release, then let me add the GitHub plugin

File: .releaserc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    [
+      "@semantic-release/changelog",
+      {
+        "changelogFile": "CHANGELOG.md"
+      }
+    ],
+    [
+      "@semantic-release/github",
+      {
+        "assets": [{ "path": "dist", "label": "package distribution" }]
```
Comment (agentmilindu) [inline .releaserc:15]: how about adding the `addReleases` also?

Comment (niloysikdar) [reply .releaserc:15]: Sounds cool, what should be the value for `addReleases` ? Can be false, "bottom" or "top"

Comment (niloysikdar) [reply .releaserc]: added the config as "addReleases": "top"

**Pull request merged**